python/tests/test_utils.py

```diff
@@ -0,0 +1,275 @@
+
+    def test_write_operations_raise_after_wrapper_closed(self):
+        """Write/flush/seek operations raise ValueError after wrapper closed."""
+        base = io.BytesIO()
+        wrapper = NonClosingStream(base)
+
+        wrapper.write(b"initial")
+        self.assertEqual(base.getvalue(), b"initial")
+
+        wrapper.close()
+
+        # Write raises after close
+        with self.assertRaises(ValueError) as ctx:
+            wrapper.write(b"after close")
+        self.assertIn("closed", str(ctx.exception).lower())
+
+        # Flush raises after close
+        with self.assertRaises(ValueError) as ctx:
+            wrapper.flush()
+        self.assertIn("closed", str(ctx.exception).lower())
+
+        # Seek raises after close
+        with self.assertRaises(ValueError) as ctx:
+            wrapper.seek(10)
+        self.assertIn("closed", str(ctx.exception).lower())
+
+        # Tell raises after close
+        with self.assertRaises(ValueError) as ctx:
+            wrapper.tell()
+        self.assertIn("closed", str(ctx.exception).lower())
+
+    def test_read_operations_raise_after_wrapper_closed(self):
+        """Read operations raise ValueError after wrapper closed."""
+        base = io.BytesIO(b"test data")
+        wrapper = NonClosingStream(base)
+
+        # Read works before close
+        data = wrapper.read(4)
+        self.assertEqual(data, b"test")
+
+        wrapper.close()
+
+        # Read raises after close
+        with self.assertRaises(ValueError) as ctx:
+            wrapper.read()
+        self.assertIn("closed", str(ctx.exception).lower())
+
+    def test_capability_checks_raise_after_wrapper_closed(self):
+        """readable/writable/seekable/isatty raise ValueError after wrapper closed."""
+        base = io.BytesIO(b"test")
+        wrapper = NonClosingStream(base)
+
+        # Before close - these work
+        self.assertTrue(wrapper.readable())
+        self.assertTrue(wrapper.writable())
+        self.assertTrue(wrapper.seekable())
+        self.assertFalse(wrapper.isatty())
+
+        wrapper.close()
+
+        # After close - these raise ValueError
+        with self.assertRaises(ValueError) as ctx:
+            wrapper.readable()
+        self.assertIn("closed", str(ctx.exception).lower())
+
+        with self.assertRaises(ValueError) as ctx:
+            wrapper.writable()
+        self.assertIn("closed", str(ctx.exception).lower())
+
+        with self.assertRaises(ValueError) as ctx:
+            wrapper.seekable()
+        self.assertIn("closed", str(ctx.exception).lower())
+
+        with self.assertRaises(ValueError) as ctx:
+            wrapper.isatty()
+        self.assertIn("closed", str(ctx.exception).lower())
+
+    def test_context_manager_closes_wrapper_only(self):
+        """Using wrapper as context manager closes wrapper but not base."""
+        base = tempfile.NamedTemporaryFile(delete=False)
+        try:
+            with NonClosingStream(base) as wrapper:
```

Comment:
Here's a big fat BUG;
The issue is that you didn't run the tests before submitting the PR, as otherwise you'd see this BUG. The test is designed to fail.